### PR TITLE
Cleans up Thief's Double Attack item bonuses

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: N/A

* **Server Mode**: Pre-renewal and Renewal

* **Description of Pull Request**: 
  * Items that grant TF_DOUBLE now no longer require bDoubleRate.
  * Adds pc_checkskill_flag() to return a skill's flag value.
Thanks to @eppc0330 and @secretdataz!

### CODE REVIEW

**Comment by eppc0330** [comment]
I think this PR will make all doubleattack items check character's TF_DOUBLE skill level right? But some items do not check it. I think defaulting this is kind of dangerous.

**Comment by eppc0330** [comment]
For example if assassin character, using katar, learning 10lv TF_DOUBLE, equips Snakehead hat (ID 5388), character should have 35% double rate, not 70%, because the item doesnt check character's TF_DOUBLE skill level.

**Comment by aleos89** [comment]
We don't have to check for the level since `pc_skill()` which is called from the `buildin_skill` function automatically detects the level learned and whether to replace it or not.

**Comment by aleos89** [comment]
> For example if assassin character, using katar, learning 10lv TF_DOUBLE, equips Snakehead hat (ID 5388: Grants 5lv TF_DOUBLE without skill level check), character should have 35% double rate, not 70%, because the item doesnt check character's TF_DOUBLE skill level.

Do you know of other items that grant skills that behave like this, or is it only select to TF_DOUBLE?

**Comment by eppc0330** [comment]
@aleos89 Quite a lot items behave like that but I will mention only ones in case of thief can equip, because in normal case other jobs else theif do not have TF_DOUBLE.  Nagan(ID 1130, not ID 13407) and snakehead hat(ID 5388), two items behave as I told you. In case of items not from KRo, sorry I have no idea because my information is based on KRo lub file.

**Comment by secretdataz** [comment]
@eppc0330 From my test on kRO, equipping a stalker (double attack level 10, 70% proc chance) with a chick hat (double attack level 2, 21% proc chance) procs double attack like a lot it's hard to believe it's just 21% chance.

**Comment by eppc0330** [comment]
> @eppc0330 From my test on kRO, equipping a stalker (double attack level 10, 70% proc chance) with a chick hat (double attack level 2, 14% proc chance) procs double attack like a lot it's hard to believe it's just 14% chance.
> 
> https://media.discordapp.net/attachments/120596466661195778/991052756154474496/Ragexe_RL9JvBM7S0.gif

WTF really? Its nonsense that iteminfo already says double rate will follow character's TF_DOUBLE level.
If you are true than it must be a bug.

**Comment by eppc0330** [comment]
@secretdataz 
But your screenshot...
A character hits doubleattack 8 times during 13 attacks.
It is 61%. No matter how small the number of trials, your screenshot shows 14% is literally nonsense. 
Are you really sure...? 

I have experience of using chick hat when I was playin assassin before and I remember it works well. Also Idk if you read Korean but KRo chick hat description says double rate will follow character's TF_DOUBLE level. 
Sorry for having doubts about your test but I need more explain because your clue(the screenshot) does not back up your opinion.

**Comment by secretdataz** [comment]
> @secretdataz But your screenshot... A character hits doubleattack 7times during 12attacks. It is 58%. No matter how small the number of trials, your screenshot shows 14% is literally nonsense. Are you really sure...?
> 
> I have experience of using chick hat when I was playing assassin before and I remember it works well. Also Idk if you read Korean but KRo chick hat description says double rate will follow character's TF_DOUBLE level. Sorry for having doubts about your test but I need more explain because your clue(the screenshot) does not back up your opinion.

I attacked the egg over 100 times. It definitely procced more than 50 times so I'm sure it's more than 14%.  
Also, chick hat's Korean description seems to be more elaborate than other double attack item for some reason.

Chick Hat's description states that.
```더블어택 2레벨 사용 가능. 더 높은 레벨을 배우고 있을 시, 해당 레벨로 발동.```

While Poison Forged Spear's description only states.
```더블어택 5Lv 사용 가능.```

But both items are scripted in the same way on official servers, only granting the skill via `EnableSkill`. This applies to every item that grants double attack. They are all scripted this way. So it's reasonable to expect them all to behave the same.

Example official scripts with irrelevant statements removed:
```
item  Poison_Forged_Spear								
	event OnStartEquip:										
		EnableSkill TF_DOUBLE 5		
```

```
item Nagan
	event OnStartEquip:
		EnableSkill TF_DOUBLE 5
```

```
item Side_Winder_Card
	event OnStartEquip:
		EnableSkill TF_DOUBLE 1
```

```
item Chick_Hat
	event OnStartEquip:
		EnableSkill TF_DOUBLE 2
```
```
item Snake_Head
	event OnStartEquip:
		EnableSkill TF_DOUBLE 5
```

Also a longer test, for larger sample size is here, https://www.youtube.com/watch?v=JzY3Ahs4EbI.

In conclusion, I believe, in all cases, unless stated otherwise, double attack with higher level learned by the player should takes a higher priority.

**Comment by aleos89** [comment]
Will also mention #3020, as it has also been confirmed on previously to take into account the player's learned level versus the granted level.

**Comment by eppc0330** [comment]
@secretdataz Oh I think I missinderstood you. I mean chick hat should have 70% chance if char has 10lv TF_DOUBLE  so you are definitely right. 

I thought you are saying chick hat grants only 14% chance sorry my English was bad.
